test_api.py

```diff
@@ -1880,13 +1891,71 @@ def test_invalid_data(self):
         patch_doc = {"display_name": "branch_id_test"}
         host_id_lists = ["notauuid", f"{self.added_hosts[0].id},notauuid"]
         for host_id_list in host_id_lists:
             with self.subTest(host_id_list=host_id_list):
                 self.patch(f"{HOST_URL}/{host_id_list}", patch_doc, 400)
 
+    def _base_patch_produces_update_event_test(self, emit_event, headers, expected_request_id):
+        patch_doc = {"display_name": "patch_event_test"}
+        host_to_patch = self.added_hosts[0].id
+
+        with patch("app.queue.egress.datetime", **{"now.return_value": self.now_timestamp}):
+            self.patch(f"{HOST_URL}/{host_to_patch}", patch_doc, 200, extra_headers=headers)
+
+        expected_event_message = {
+            "type": "updated",
+            "host": {
+                "account": self.added_hosts[0].account,
+                "ansible_host": self.added_hosts[0].ansible_host,
+                "bios_uuid": self.added_hosts[0].bios_uuid,
+                "created": self.added_hosts[0].created,
+                "culled_timestamp": (
+                    dateutil.parser.parse(self.added_hosts[0].stale_timestamp) + timedelta(weeks=2)
+                ).isoformat(),
+                "display_name": "patch_event_test",
+                "external_id": self.added_hosts[0].external_id,
+                "fqdn": self.added_hosts[0].fqdn,
+                "id": self.added_hosts[0].id,
+                "insights_id": self.added_hosts[0].insights_id,
+                "ip_addresses": self.added_hosts[0].ip_addresses,
+                "mac_addresses": self.added_hosts[0].mac_addresses,
+                "reporter": self.added_hosts[0].reporter,
+                "rhel_machine_id": self.added_hosts[0].rhel_machine_id,
+                "satellite_id": self.added_hosts[0].satellite_id,
+                "stale_timestamp": self.added_hosts[0].stale_timestamp,
+                "stale_warning_timestamp": (
+                    dateutil.parser.parse(self.added_hosts[0].stale_timestamp) + timedelta(weeks=1)
+                ).isoformat(),
+                "subscription_manager_id": self.added_hosts[0].subscription_manager_id,
+                "system_profile": {},
+                "tags": [
+                    {"namespace": "no", "key": "key", "value": None},
+                    {"namespace": "NS1", "key": "key1", "value": "val1"},
+                    {"namespace": "NS1", "key": "key2", "value": "val1"},
+                    {"namespace": "SPECIAL", "key": "tag", "value": "ToFind"},
+                ],
+                "updated": self.added_hosts[0].updated,
+            },
+            "platform_metadata": None,
+            "metadata": {"request_id": expected_request_id},
+            "timestamp": self.now_timestamp.isoformat(),
+        }
+
+        emit_event.assert_called_once()
+        event_message = json.loads(emit_event.call_args[0][0])
+        self.assertEqual(event_message, expected_event_message)
```

Comment:
Let’s test the _key_ too and also the headers once added.

```suggestion
        emitted_event = emit_event.call_args[0]
        event_message = json.loads(emitted_event[0])
        self.assertEqual(event_message, expected_event_message)
        self.assertEqual(emitted_event[1], self.added_hosts[0].id)
        self.assertEqual(emitted_event[2], "updated")
```